src/org/skypro/skyshop/basket/ProductBasket.java

```diff
@@ -45,9 +43,11 @@ public void printProductBasket() {
     }
 
     public boolean searchProduct(String name) {
-        for (Product product : productBasket) {
-            if (product != null && product.getName().equalsIgnoreCase(name)) {
-                return true;
+        for (Map.Entry<String, LinkedList<Product>> product : productBasket.entrySet()) {
+            for (Product p : product.getValue()) {
+                if (p.getName().equalsIgnoreCase(name)) {
+                    return true;
+                }
             }
         }
         return false;
```

Comment:
это всё заменяется на 1 строчку кода
```suggestion
        return productBasket.containsKey(name);
```